### PR TITLE
Update KeyVault(Keys,Certificates) README

### PR DESCRIPTION
**Pull request by tzhanl**
Purpose : Update KeyVault-keys and KeyVault-certificates README based on Preview4
Total changes : 5

### 1. Necessary parameters "updated" and "purge"  missed of CLI command

**Original**
_According this README , it request purge and update permission but this command don’t have those three parameters_

image

**Proposal**

image

### 2. Missing hyper link of Package(NuGet)

**Original**

image

Package(NuGet) link : https://www.nuget.org/packages/Azure.Security.KeyVault.Certificate/

**Proposal**

Package(NuGet) link : https://www.nuget.org/packages/Azure.Security.KeyVault.Certificates/

### 3. Standard keyvault don’t support import hardware key
_Will throw an exception while HSM value is true , because standard KeyVault don’t support import hardware key_

**Original**

image

**Proposal**
_Guess we can add follow comments_

image

### 4. Necessary parameters "encrypt" ,"decrypt" and "Update" missed of CLI command
_According this README , it request encrypt , decrypt and update permission but this command don’t have those three parameters_

**Original**

image

**Proposal**

image

### 5.Missed content of List Certificates

**Original**

image

**Proposal**

image

### CODE REVIEW

File: sdk/keyvault/Azure.Security.KeyVault.Keys/README.md [hunk 1]
```diff
@@ -115,113 +115,115 @@ The following section provides several code snippets using the [above created](#
...
+```C# CreateKey
+// Create a key. Note that you can specify the type of key
+// i.e. Elliptic curve, Hardware Elliptic Curve, RSA
+Key key = client.CreateKey("key-name", KeyType.Rsa);
+
+Console.WriteLine(key.Name);
+Console.WriteLine(key.KeyMaterial.KeyType);
+
+// Create a software RSA key
+var rsaCreateKey = new RsaKeyCreateOptions("rsa-key-name", hsm: false);
```
Comment (heaths) [inline sdk/keyvault/Azure.Security.KeyVault.Keys/README.md]: Do not modify the code in the README directly. Instead, modify the code in associated samples/SampleSnippets.cs that has #regions with the same name as the text here after "c#", then run a ps1 script in the root "eng" or "tools" folder, something like Update-Snippets.ps1.

File: sdk/keyvault/Azure.Security.KeyVault.Keys/README.md [hunk 2]
```diff
@@ -115,113 +115,115 @@ The following section provides several code snippets using the [above created](#
...
+Key rsaKey = await client.CreateRsaKeyAsync(rsaCreateKey);
+
+Console.WriteLine(rsaKey.Name);
+Console.WriteLine(rsaKey.KeyMaterial.KeyType);
+
+// Create a hardware Elliptic Curve key
```
Comment (heaths) [inline sdk/keyvault/Azure.Security.KeyVault.Keys/README.md]: Probably should include the same note about needing a premium SKU here as you did above.

**Review by heaths — CHANGES_REQUESTED** [review]

File: sdk/keyvault/Azure.Security.KeyVault.Certificates/README.md [hunk 3]
```diff
@@ -138,6 +138,18 @@ Console.WriteLine(deletedCert.ScheduledPurgeDate);
 await client.PurgeDeletedCertificateAsync("MyCertificate");
 ```
 
+### List Certificates
+`GetCertificatesAsync` Enumerates the certificates in the vault, returning select properties of the
+certificate, sensative feilds of the certificate will not be returned. This operation
```
Comment (heaths) [inline sdk/keyvault/Azure.Security.KeyVault.Certificates/README.md]: "sensitive fields". Please be sure to run spell check. If doing this in Code, there are a few good extensions you can use.

Also, this line should be "certificate. Sensitive fields...". That's a new sentence.

File: sdk/keyvault/Azure.Security.KeyVault.Keys/README.md [hunk 4]
```diff
@@ -132,7 +132,7 @@ Console.WriteLine(rsaKey.KeyMaterial.KeyType);
 
 // Create a hardware Elliptic Curve key
 // Because only premium key vault supports HSM backed keys , please ensure your key vault
-// SKU is premium while you set "hsm" value to true.
+// SKU is premium while you set "hsm" value to true
```
Comment (heaths) [inline sdk/keyvault/Azure.Security.KeyVault.Keys/README.md]: "while" -> "when".

File: sdk/keyvault/Azure.Security.KeyVault.Keys/samples/SampleSnippets.cs [hunk 5]
```diff
@@ -67,6 +67,8 @@ public void CreateKey()
             Console.WriteLine(rsaKey.KeyMaterial.KeyType);
 
             // Create a hardware Elliptic Curve key
+            // Because only premium key vault supports HSM backed keys , please ensure your key vault
+            // SKU is premium while you set "hsm" value to true
```
Comment (heaths) [inline sdk/keyvault/Azure.Security.KeyVault.Keys/samples/SampleSnippets.cs]: "while" -> "when" in these other places too.

**Review by heaths — APPROVED** [review]

File: sdk/keyvault/Azure.Security.KeyVault.Certificates/README.md [hunk 6]
```diff
@@ -140,7 +140,7 @@ await client.PurgeDeletedCertificateAsync("MyCertificate");
 
 ### List Certificates
 `GetCertificatesAsync` Enumerates the certificates in the vault, returning select properties of the
-certificate, sensative feilds of the certificate will not be returned. This operation
+certificate. Sensitive feilds of the certificate will not be returned. This operation
```
Comment (heaths) [inline sdk/keyvault/Azure.Security.KeyVault.Certificates/README.md]: "fields". That's still spelled wrong.

**Review by heaths — APPROVED** [review]

**Review by heaths — COMMENTED** [review]
@weshaggard or @chidozieononiwu, if binary files are being updated regularly, should we look into enabling LFS? From experience, at least, I recommended opting all binary extensions in (or at least known), then explicitly filtering out existing binaries that seldom-changed and get used a lot (switching between branches with/without LFS enabled causes issues, though we do pretty much have a forward-only practice now).

**Review by pakrym — CHANGES_REQUESTED** [review]
samples.zip should be removed from the root of the repo.

File: sdk/keyvault/Azure.Security.KeyVault.Certificates/README.md [hunk 7]
```diff
@@ -138,6 +138,20 @@ Console.WriteLine(deletedCert.ScheduledPurgeDate);
 await client.PurgeDeletedCertificateAsync("MyCertificate");
 ```
 
+### List Certificates
+`GetCertificatesAsync` Enumerates the certificates in the vault, returning select properties of the
```
Comment (heaths) [inline sdk/keyvault/Azure.Security.KeyVault.Certificates/README.md]: Should be "enumerates", and we also don't explicitly mention the "Async" suffix for methods unless necessary. This is a naming convention in .NET but we refer to the method names as more of a family, i.e. "GetCertificates" for both `GetCertificates` and `GetCertificatesAsync` - they are sync/async alternatives to each other.

**Review by heaths — APPROVED** [review]
I made a couple of tweaks, but otherwise good to go.

**Comment by heaths** [comment]
Talking with Wes and Dozier, we noticed it was in the root so I removed it.

**Review by pakrym — APPROVED** [review]

**Pull request merged**